src/client/d3/d3.js

```diff
@@ -1,49 +1,82 @@
-const { width, height, jsonUrl, svg } = require('./setup.js');
-const { sortWithMax, binByTag, memoryNodesAndLinks, centralMaxNodesByTag } = require('../node_transformations');
+const { openTagMenu, showDeleteButton, hoveringOnDelete, hideDeleteButton, submitNewMemory, tagSorting } = require('../helpers/helpers.js');
+const { width, height, jsonUrl, svg, fdGrp, nodeGrp, linkGrp } = require('./setup.js');
+const { sortWithMax, binByTag, centralMaxNodesByTag, memoryNodesAndLinks } = require('../node_transformations');
 const { appendPopUp, randomPopUp } = require('./modals.js');
 
 const url = location.hostname ? '/memories' : jsonUrl;
 
 d3.json(url, (err, data) => {
-  // binByTag sorts data by tag
-  // e.g. {family: Array(5), pets: Array(5), friends: Array(5)}
+  formatData(data, render);
+  openTagMenu();
+  submitNewMemory();
+});
+
+const formatData = (data, callback) => {
+// binByTag sorts data by tag
+// e.g. {family: Array(5), pets: Array(5), friends: Array(5)}
   const binnedByTag = binByTag(data);
   // sortedWithMax sorts each tag group to separate max memory (by likes) from others in its group
   const sortedWithMax = [];
   Object.keys(binnedByTag).forEach((tagKey) => {
     sortedWithMax.push(sortWithMax(binnedByTag[tagKey]));
   });
-  // taggedNodesByTag returns an object with the cx and cy for the central node within each tag group
+// taggedNodesByTag returns an object with the cx and cy for the central node within each tag group
   const centralNodesByTag = centralMaxNodesByTag(sortedWithMax, 160, 120);
 
-  // Add unique tags to tag list for user to select from
+// Add unique tags to tag list for user to select from
   Object.keys(centralNodesByTag).forEach((tag) => {
     tag = tag.replace(/\W/g, '');
     $('.tags').append(
-      `<li class='tag-container ${tag}'>
-        <p class='tagLabel'>${tag}</p>
-        <img class='filter-tags ${tag}' src="./assets/icons/navigate/close_icon.svg"/>
-      </li>`);
-  });
-  $('.tags').append(`
-    <li class='clear-tags'>clear</li>
-    <li class='close-tags'>
-      <img class='close-icon' src="./assets/icons/navigate/close_icon.svg">
-      </img>
+    `<li class='tag-container ${tag}'>
+      <p class='tagLabel'>${tag}</p>
+      <img class='filter-tags ${tag}' src="./assets/icons/navigate/close_icon.svg"/>
     </li>`);
-  // processedData returns a list of nodes and links
+  });
+
+  $('.tags').append(
+  `<li class='clear-tags'>clear</li>
+  <li class='close-tags'>
+    <img class='close-icon' src="./assets/icons/navigate/close_icon.svg">
+    </img>
+  </li>`);
+
+// processedData returns a list of nodes and links
   const processedData = memoryNodesAndLinks(centralNodesByTag, sortedWithMax);
 
   const nodeDataArray = [];
   Object.keys(processedData.nodes).forEach((key) => {
     nodeDataArray.push(processedData.nodes[key]);
   });
+  callback(processedData, nodeDataArray);
```

Comment:
This function is not asynchronous, you can return values rather than have a callback